### PR TITLE
Fixes #2013. Starting having MainLoopTests.InvokeLeakTest failure.

### LINKED ISSUE #2013: Starting having MainLoopTests.InvokeLeakTest failure after push develop branch to my fork .
```
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
  Failed Terminal.Gui.Core.MainLoopTests.InvokeLeakTest [1 s]
  Error Message:
[xUnit.net 00:01:04.81]     Terminal.Gui.Core.MainLoopTests.InvokeLeakTest [FAIL]
   System.TimeoutException : Timeout: Increment lost. tbCounter (41) didn't change after waiting 500 ms. Failed to reach 10000 on pass 1
  Stack Trace:
     at Terminal.Gui.Core.MainLoopTests.RunTest(Random r, TextField tf, Int32 numPasses, Int32 numIncrements, Int32 pollMs) in /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/MainLoopTests.cs:line 556
   at Terminal.Gui.Core.MainLoopTests.<>c__DisplayClass24_0.<InvokeLeakTest>b__0() in /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/MainLoopTests.cs:line 577
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at Terminal.Gui.Core.MainLoopTests.InvokeLeakTest() in /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/MainLoopTests.cs:line 582
--- End of stack trace from previous location ---

Attachments:
  /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/TestResults/5774b41c-7d1a-461a-a399-f2edfe5328aa/coverage.opencover.xml
Failed!  - Failed:     1, Passed:   873, Skipped:     0, Total:   874, Duration: 1 m 9 s - /home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/bin/Debug/net6.0/UnitTests.dll (net6.0)
     4>Done Building Project "/home/runner/work/Terminal.Gui/Terminal.Gui/UnitTests/UnitTests.csproj" (VSTest target(s)) -- FAILED.
     1>Done Building Project "/home/runner/work/Terminal.Gui/Terminal.Gui/Terminal.sln" (VSTest target(s)) -- FAILED.

Build FAILED.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:01:19.40
Error: Process completed with exit code 1.

```

### PR DESCRIPTION
**Pull request by BDisp**
Fixes #2013 - Trying to fix fails unit test.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

File: UnitTests/ApplicationTests.cs [hunk 1]
```diff
@@ -1301,7 +1301,7 @@ public void TestAddManyTimeouts ()
 					var myi = i;
 
 					Task.Run (() => {
-						Task.Delay (100).Wait ();
+						Thread.Sleep (100);
```
[the remark this answers is not in the document]
Comment (BDisp) [reply UnitTests/ApplicationTests.cs]: Right. Done on #2018. Thanks.

File: UnitTests/ApplicationTests.cs [hunk 2]
```diff
@@ -1318,7 +1318,7 @@ public void TestAddManyTimeouts ()
 						if (myi == 0) {
 
 							// let the timeouts run for a bit
-							Task.Delay (5000).Wait ();
+							Thread.Sleep (5000);
```
[the remark this answers is not in the document]
Comment (BDisp) [reply UnitTests/ApplicationTests.cs]: Right. Done on #2018. Thanks.

File: UnitTests/MainLoopTests.cs [hunk 3]
```diff
@@ -541,25 +541,27 @@ private static void Launch (Random r, TextField tf)
...
-				{
-					var tbNow = tbCounter;
-					Thread.Sleep (pollMs);
-					if (tbCounter == tbNow) {
-						// No change after sleep: Idle handlers added via Application.MainLoop.Invoke have gone missing
-						Application.MainLoop.Invoke (() => Application.RequestStop ());
-						throw new TimeoutException (
-							$"Timeout: Increment lost. tbCounter ({tbCounter}) didn't " +
-							$"change after waiting {pollMs} ms. Failed to reach {(j + 1) * numIncrements} on pass {j + 1}");
+			Task.Run (() => {
```
[the remark this answers is not in the document]
Comment (BDisp) [reply UnitTests/MainLoopTests.cs]: I understand. Thanks for your help. The suggested fix in https://github.com/gui-cs/Terminal.Gui/issues/2013#issuecomment-1248908545 really fix it.

**Review by tig — APPROVED** [review]
LGTM

**Pull request merged**